version: '3.8'

services:
  traefik:
    image: "traefik:v2.10.4"
    container_name: "traefik"
    restart: always
    ports:
      - "80:80"
      - "443:443"
      - "8080:8080"
    command:
      - --log.level=DEBUG
    networks:
      - external
    environment:
      - CF_API_EMAIL= ${CLOUDFLARE_API_EMAIL}
      - CF_API_KEY= ${CLOUDFLARE_API_KEY}
    volumes:
      - /etc/localtime:/etc/localtime:ro
      - /var/run/docker.sock:/var/run/docker.sock:ro
      - ./config/traefik/traefik.yml:/traefik.yml:ro
      - ./config/traefik/data/:/data/
      - ./config/traefik/config.yml:/config.yml

  portainer:
    container_name: portainer
    image: portainer/portainer-ce:latest
    command: -H unix:///var/run/docker.sock 
    restart: always
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
      - portainer_data:/data
    ports:
      - 9000:9000
    networks:
      - external

volumes:
  minio-data:
  portainer_data:

networks:
  external:
  internal:
 